applications/tari_base_node/src/grpc.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright 2019. The Tari Project
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+use crate::builder::NodeContainer;
+use base_node_grpc::*;
+use log::*;
+use prost_types::Timestamp;
+use tari_core::{
+    base_node::{comms_interface::CommsInterfaceError, LocalNodeCommsInterface},
+    blocks::BlockHeader,
+    mempool::service::LocalMempoolService,
+    proof_of_work::PowAlgorithm,
+};
+use tari_crypto::tari_utilities::{epoch_time::EpochTime, ByteArray, Hashable};
+use tokio::runtime;
+use tonic::{transport::Server, Code, Request, Response, Status};
+
+const LOG_TARGET: &str = "base_node::grpc";
+
+pub mod base_node_grpc {
+    tonic::include_proto!("tari.base_node");
+}
+
+pub struct BaseNodeGrpcServer {
+    executor: runtime::Handle,
```

Comment:
FYI We're moving towards using tokio's annotations or grabbing the handle from `tokio::runtime::Handle::current()`; which saves us having to pass an executor or handle around everywhere